### ISSUE TITLE
FlashAttention actually does not support attention mask

### ISSUE THREAD

**Issue by HJoonKwon** [cvg/LightGlue#116]
Thanks for your great work!

I'm just curious whether your code [here](https://github.com/cvg/LightGlue/blob/main/lightglue/lightglue.py#L113) is using flash or not when mask is not ```None```. My guess is it's using memory efficient attention instead since PyTorch flash attention kernel does not support attention mask. Thank you!

**Comment by Phil26AT** [comment]
Hey @HJoonKwon! Damn, very good find, thank you! I guess this does matter in compiled forward, where we are padding inputs to static dimensions. We'd need to run the benchmarks, but maybe avoiding the call to `half()` could improve throughput then.

**Comment by HJoonKwon** [comment]
Great! Thank you again for your great work. I got inspired a lot.

**Comment by LudvigDillen** [comment]
On the topic of FlashAttention, you link to FlashAttention and not FlashAttention2 here
![image](https://github.com/cvg/LightGlue/assets/70961163/2fbe59e3-9406-4f1b-8219-81f1edaf1b29)
Isn't the second version used? If not, why? Seems quite much faster
![image](https://github.com/cvg/LightGlue/assets/70961163/11fe8f91-a099-4970-94b2-7ab858302bf6)

FlashAttention: https://arxiv.org/abs/2205.14135
FlashAttention2: https://arxiv.org/pdf/2307.08691.pdf?trk=public_post_comment-text
